### ISSUE TITLE
Reaction emoji disappears/reappears after quickly adding/removing reactions

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16506]
# Rollout Plan
- [x] #18427
- [x] https://github.com/Expensify/Expensify/issues/281934
- [x] #19531
- [x] #19529
- [x] #19534
- [x] #19535

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to any chat
2. Type a text
3. Long press and react to some emojis , Please do select 3 / 4 emojis ( using add emoji button)
4. Notice that you have now multiple emoji reactions on that typed text.
5.  Now unselect those emojis

    Notice that after unselecting all emojis, they reappear

## Expected Result:
After unselecting emojis, they should not reappear

## Actual Result:
When all emojis are unselected, usually one or all emojis are restored. Further, if new text is typed below. Such restoration doesn’t last.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native ( Device model : one plus nord)
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.87-1
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/227727393-9e12dfb6-a7a9-403b-9feb-ffb7fc8e72a5.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ashimsharma10
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679675126743679

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016e24bec63a2e084f</li>
        <li>Upwork Job ID: 1640357797486551040</li>
        <li>Last Price Increase: 2023-03-27</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @eVoloshchak (`Internal`)

**Comment by stitesExpensify** [comment]
This is the same as number 3 from my slack post here https://expensify.slack.com/archives/C049HHMV9SM/p1678286500121729

I'll leave this open and assigned to me since we didn't have an issue for it, but it is known

**Comment by priyeshshah11** [comment]
@stitesExpensify Just FYI, this also happens when you add/remove reactions when offline & then go back online as it effectively is the same as reacting quickly as both requests will be sent in quick successions.

**Comment by laurenreidexpensify** [comment]
@stitesExpensify what is the latest on this one?

**Comment by stitesExpensify** [comment]
No update yet. This is a bug that is not specific to emojis and happens in other places like pins as well, so it will require a larger discussion. I've been focusing on the smaller emoji reaction polish, and plan on coming back to this next week when I have more time.

**Comment by MelvinBot** [comment]
@eVoloshchak @stitesExpensify @laurenreidexpensify this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@eVoloshchak, @stitesExpensify, @laurenreidexpensify Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by laurenreidexpensify** [comment]
@stitesExpensify is this merged now?

**Comment by laurenreidexpensify** [comment]
Note for myself: Haven't hired anyone in Upwork yet, waiting for confirmation that we're fixing this.

**Comment by stitesExpensify** [comment]
No, this has not been dealt with. I'm going to try to create a plan today. This issue is pretty complicated due to the interaction between optimistic onyx data, the server response, and the pusher data, so I need to figure out the best way to prevent regressions

**Comment by laurenreidexpensify** [comment]
@stitesExpensify any update on the plan?

**Comment by stitesExpensify** [comment]
There are 2 options as far as I can tell. 

1. Don't send pusher updates to the client that reacted, but do send failure data
    - Pros:
        - Directly solves the problem of your own reactions causing flickering
        - No front end trickery
    - Cons:
        - As we saw with inviting to large group chats, deciding who to send pusher data to can be slow
        - Goes against the pattern of always pushing data back to the front end for writes
2. Always prefer the local changes for the user instead of accepting the server push
    - Pros:
        - This will prevent flickering for both yourself, and multiple users reacting at once
    - Cons:
        - Involves manipulating server data on the front end which is against our development philosophy
        - Uses the front end as source of truth instead of the server
        - Could cause problems with adding/removing emojis on other clients (how do we know when it's legit or not)
3. Some combination of both of the above (which might be necessary)


I plan on starting with #1 and testing it out because it is the most straightforward, and then I'll come back for 2/3 if necessary

**Comment by MelvinBot** [comment]
@eVoloshchak @stitesExpensify @laurenreidexpensify this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by laurenreidexpensify** [comment]
@stitesExpensify listed a plan above that we'll work on this week

**Comment by stitesExpensify** [comment]
So I almost immediately realized a problem with `1` which is that we still need to use pusher for the user that reacted, because we need to push out the reaction to the user's other devices. 

I'm thinking the better way to do it will be to control the merge somehow. Looking into it today, this is my top priority now btw

**Comment by MelvinBot** [comment]
@eVoloshchak, @stitesExpensify, @laurenreidexpensify Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@eVoloshchak @stitesExpensify @laurenreidexpensify this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by laurenreidexpensify** [comment]
@stitesExpensify any update? thanks!

**Comment by stitesExpensify** [comment]
I'm at a bit of a loss on this one tbh, my attempts to fix it haven't been successful. I've started asking around to see how (or if) we fixed this same issue with pinning reports to see if the same solution could be applied here.

**Comment by stitesExpensify** [comment]
The pinning issue is not fixed FWIW. 

Another potential solution would be to basically batch all of the requests into a single API call with multiple emojis so that it all gets sent at once. 

Some potential issues with that are:
1. If you are the first to react and do a bunch of reactions when someone else does one, you will get a push from someone else and you will now be the _second_ person to react (your order will switch)
2. We will still have the same problem where if you send the batch and then react one more time right after, the flicker will still be there

**Comment by laurenreidexpensify** [comment]
Discussing with @tgolen [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1682371176391969)

**Comment by tgolen** [comment]
Yesterday a built a POC concept in NewDot for moving reactions to their own Onyx collection and stored the data as objects instead of arrays. This works well on the front-end. Today, I'm going to attempt to build that into the PHP API, and then eventually the Auth commands.

**Comment by tgolen** [comment]
OK, I think I've got an idea of how this is going to work. To start with, here is the data format that I'm looking at using. Are we OK with this @stitesExpensify @marcaaron @hannojg ?

```
message = {
    "html": "test message with emoji \ud83d\ude1c",
    "reactionObjects": {
        "<emojiCode>": {
            "createdAt": 123456789,
            "users": {
               "<accountID>": {
                    "createdAt": 123456789,
                    "skinTone": 1,
                },
            },
        },
    },
}
```

A few notes for completeness:
- The new property is called `reactionObjects`. I am not in love with this, but I couldn't think of any other way to migrate to a new data format without using a new property name. I'm open to ideas and suggestions here!
- The `emojiCode` and `accountID` will now be the object keys
- The `createdAt` timestamp will come from the front-end and it will be used in the UI to ensure that the emojis and users are sorted in the order of creation
- I don't really think we need to consider a migration from the old format. This means we will lose all previous reactions, but I think that it is fine at this stage and I don't think trying to keep them is very high priority. At some point, we could maybe delete them to clean up the database, but it's a minimal amount of data IMO

**Comment by marcaaron** [comment]
Did we consider the tradeoffs of letting the format on the server continue to be an array and transform it into an object when handing off to the front end?

As for an alternative name I can offer... `userReactions`, `participantReactions` or `emojiReactions`

**Comment by stitesExpensify** [comment]
I think that the new data format makes sense!

**Comment by tgolen** [comment]
I like `emojiReactions` better than `reactionObjects`. @stitesExpensify do you have a preference?

**Comment by stitesExpensify** [comment]
Definitely `emojiReactions`

**Comment by tgolen** [comment]
Cool beans. I'll use that then. There was one more pieces I wanted to get some thoughts on. 

With this new way of storing the data, PHP is able to send all of the proper Onyx updates without the response from Auth. I guess my question is... should we still continue to return something from Auth (like the full `emojiReactions` object) even if it's not necessary or should we just not return anything and only make sure that PHP handles any Auth errors?

**Comment by tgolen** [comment]
@stitesExpensify @marcaaron do you have thoughts on the above about Auth sending back a response at all?

**Comment by stitesExpensify** [comment]
I think that makes sense. As long as we are handling the errors, then it doesn't seem like we need to return the full list from auth

**Comment by tgolen** [comment]
@marcaaron that will affect some of the refactoring you were doing on the Auth commands since you were changing what the returned response would be.

I'm trying to think about how to roll this out with a minimal amount of breakage. This is what I'm thinking:

1. Roll out Auth so that both data formats are stored and updated in the existing commands
2. Add a parameter to the PHP endpoints for add/remove reactions called `useObjectDataFormat`. When it's `true` then it will process and return the new data format to NewDot
3. Update NewDot to use the new data format and pass `useObjectDataFormat` as `true`
4. Once ^ is deployed, remove the `useObjectDataFormat` parameter from PHP
5. Once ^ is deployed, remove the `useObjectDataFormat` parameter from NewDot
6. Finally, remove the code in Auth which is storing the old array data format

**Comment by marcaaron** [comment]
ah tbh, I don't know if I fully understood the question 😓 

>PHP is able to send all of the proper Onyx updates without the response from Auth

Which updates are we talking about specifically and how does PHP able to send anything without something from Auth? 🤔

**Comment by tgolen** [comment]
It's basically the same "optimistic" principle that we use on the front-end. In the case of reactions, PHP knows exactly what to send to update Onyx because it's a super simple merge() with the changes to the reactions. PHP doesn't need anything from Auth to:

1. Add a new emoji: `Onyx.merge('reportActionsReactions_$reportID.$reportActionID', {<newEmojiCode>: {'users': {<accountID>: {skinTone: <skinTone>}}}})`
1. Add a reaction to an emoji that already exists: same as previous

**Comment by laurenreidexpensify** [comment]
@stitesExpensify @marcaaron any thoughts on the above from @tgolen ^^

**Comment by tgolen** [comment]
OK, here is my plan. I'm going to make these into separate issues so it's easier to track the PRs and progress.

- [ ] Support both existing and new emojiReactions data format in Auth
- [ ] [HOLD] Add a parameter to PHP add/remove emoji reaction commands for `useEmojiReactions`
- [ ] [HOLD] Support emojiReactions data format in NewDot
- [ ] [HOLD] Cleanup: Remove `useEmojiReactions` param from PHP
- [ ] [HOLD] Cleanup: Remove `useEmojiReactions` param from NewDot
- [ ] [HOLD] Cleanup: Remove original "reactions" data format code in Auth

**Comment by tgolen** [comment]
I submitted a PR to Auth on Friday, so it should be reviewed soon. Once it is merged, it will unblock the next step.

**Comment by tgolen** [comment]
I submitted a PR to PHP today. Once that's merged, I'll get the PR for NewDot ready.

**Comment by tgolen** [comment]
This is slowly moving forward. Things are just kind of hung up on deploys and we don't want to rush this since things have to be deployed in a specific order.

**Comment by tgolen** [comment]
I am currently working on the PHP and NewDot changes to support both the old and the new emoji formats. It will be my primary focus today

**Comment by laurenreidexpensify** [comment]
I don't think this is overdue, the weekend just confused things a bit. @tgolen continues to. be active on this.

**Comment by tgolen** [comment]
I did a lot of work on this yesterday and ran into several edge cases during testing that was giving me trouble. Hopefully, I'll be able to get the functionality working perfectly today, then I can get the PRs ready for review in PHP and NewDot.

**Comment by laurenreidexpensify** [comment]
@tgolen how did yesterday go? did you manage to get the functionality working okay?

**Comment by tgolen** [comment]
I was able to do some further testing today and got the functionality implemented to sort the reactions in the right order.

Sadly though, I still keep running into strangeness when testing:
1. Sometimes it takes the front-end 10-15 seconds between clicking on an emoji, and the API request being made. I think this has something to do with Pusher authentication taking a while. It seems to only happen in the first 10-15 seconds after loading the app. Not a big problem because the network requests do end up happening, but it's a little annoying.
2. Sometimes after removing several emoji in a row, a phantom emoji comes back (same as the original issue)
3. Sometimes it takes 5-10 seconds after receiving the pusher event that an emoji has been added/removed before the UI updates (this seems to be a new problem)
4. Pusher events are duplicated (I asked about this in Slack [here](https://expensify.slack.com/archives/C03TQ48KC/p1684362006150329))

It doesn't seem like any of these are problems on the PHP side (except for 4, so that's good). 

I can't help shake the feeling though that this is getting reactions into a state that is worse than what we had originally. The problems only seem to manifest when quickly adding/removing (even in the original case). So, it's frustrating that I haven't really been able to improve much from what we had originally.

@stitesExpensify If you had the time, could you spend maybe about 15 minutes testing this out and playing with it? You'll need to be on updated Auth, web-e and NewDot need to be on the branch `tgolen-migrate-reactions`. I've been testing it with two clients open and signed in as different users chatting with themselves. I'd like to know if you're seeing the same things I am (especially 2 & 3).

**Comment by stitesExpensify** [comment]
I've been heads down on threads, but I'll definitely try to carve out some time tomorrow to take a look!

**Comment by tgolen** [comment]
Brandon and I were both OOO today, so hopefully, we can commence testing on this. I do have a high-priority Pusher bug to look into and address today so I don't know if I will have very much time to work on this today. I would love for Brandon to try and test it out though and give it a quick run.

**Comment by tgolen** [comment]
Latest update [here](https://github.com/Expensify/Web-Expensify/pull/37193)

**Comment by tgolen** [comment]
@stitesExpensify Have you had time to test this yet? I'm getting pretty concerned with the timing of this because this is the last week I am working before my sabbatical. It stinks that I'm not going to be able to work on this for two months and I think you're going to need to take this back over from me to get it across the finish line (or find someone else who can).

I have one thought that keeps going through my head... as much as I would love to fix this, it's not a problem experienced in the real world because no one is adding/removing emojis at a speed that triggers this behavior.

Here is what I think the current problem is (read [this first to get background context](https://github.com/Expensify/Web-Expensify/pull/37193#issuecomment-1540636005)).

As I explained in that post, two users adding or removing different emojis is no longer an issue. However, the new issue is when adding/removing the same emoji. Think of these happening all at once:

Initial state of data:
```
key: {a: 1}
```

Multiple requests to add/remove one of the properties
```
Onyx.merge(key, {a: null});
Onyx.merge(key, {a: 2});
Onyx.merge(key, {a: null});
Onyx.merge(key, {a: 3});
```

Now, the final state of the data could be any of these, depending on the order of the requests:
- `{a: null}`
- `{a: 2}`
- `{a: 3}`

I highly suspect that [the fix](https://github.com/Expensify/App/issues/12775) for the "reply effect" will help us out, and I don't think I've tested it with this fix yet.

# Next Steps
- Update both web-e and app branches to latest code and see if the "replay effect" fix helps with things

**Comment by tgolen** [comment]
OK, I just tested it... and... it's surprisingly resilient now. I wasn't able to reproduce any issues with phantom reactions. @stitesExpensify would love if you can verify that also. I've updated both the App and Web-E branches to be current with `main`.

**Comment by stitesExpensify** [comment]
It seems to be working really well! Definitely better than it was. One thing I noticed while testing is that double pressing a reaction causes the old issue. This seems like it will be a decently common use case of "React, oops I didn't mean to click that one I want to undo it"

https://github.com/Expensify/App/assets/42391420/21757b47-c147-48ea-b2a0-9bd23cc10429

**Comment by tgolen** [comment]
Ah, interesting. I'm so glad your testing went well! I'll try that flow out and see what's going on. I maybe need to look more into the fix for the replay feature to see what exactly was done and how it works. It would be so nice if this API just ignored any server response and only did optimistic data (for the person clicking the reactions).

**Comment by neil-marcellini** [comment]
@tgolen 
So for some reason, I don't think the replay effect fix is working properly. NewDot is still receiving Push events for the requesting client, but they are supposed to be ignored for that client by [sending the Pusher `socket_id` in the event](https://github.com/Expensify/Web-Expensify/blob/5d68ad8cd7e77d93a4397b8587dac5d48b2e3b9e/lib/Push/Service.php#L239-L243). The pushed data is [returned in the response Onyx data](https://github.com/Expensify/Web-Expensify/blob/c71ce5100c348fc0e13ac402b6752801c9c52197/api.php#L3303-L3305) instead. So we're actually getting two identical updates when we should get one.

**Comment by stitesExpensify** [comment]
Hm I'm still getting the same problem with double clicking an emoji 😞 The rest of it seems to be working very well still


https://github.com/Expensify/App/assets/42391420/aec1861a-949f-47f3-a7e0-8c4a475d3b82

**Comment by stitesExpensify** [comment]
I also ran into another issue where the emojis are switching places, but it is very hard to reproduce and involves un-reacting right after someone else reacted. In this case, the tooltip also appears to be very broken. TBH I would say we probably don't need to worry about it for now


https://github.com/Expensify/App/assets/42391420/47152582-7d45-4d53-a714-d7ebedf11d98

**Comment by tgolen** [comment]
@stitesExpensify for the double-click issue, did you run `composer install` on the web-e PR to be sure you're using the latest version of the pusher lib?

**Comment by stitesExpensify** [comment]
Double click issue is fixed! The switching is more prevalent in this video... Trying to figure out if there are easier repro steps

https://github.com/Expensify/App/assets/42391420/50dce4a5-6d57-46aa-82eb-3445aa1ab6af

**Comment by tgolen** [comment]
Ah, you know what I think might be happening... I bet both emojis get the same timestamp (which is in seconds). Then their order could get pretty random. Maybe we could sort by timestamp + emojiName to ensure they will always be in the same order.

**Comment by tgolen** [comment]
Ah, OK. Yes, I understand why now. Good job reproducing it. Now, I just need to think of how to solve it. Anyway, this is a frontend only bug so reviewing and merging https://github.com/Expensify/Web-Expensify/pull/37193 should still be done first.

**Comment by tgolen** [comment]
@stitesExpensify I'm assigning this over to you so you can keep running with it while I am on sabbatical. I've updated the description of this issue to contain the detailed rollout plan. You just need to do everything in order from top to bottom and all the PRs are already drafted. Once it comes time to bring a PR off hold, you'll just need to merge `main` into it, solve any conflicts, and then have it reviewed and merged.

All the HOLDs are a little annoying, but it should be straight-forward.

**Comment by stitesExpensify** [comment]
Waiting for another deploy before we can move this forward

**Comment by stitesExpensify** [comment]
Waiting on a review of https://github.com/Expensify/App/pull/20087

**Comment by stitesExpensify** [comment]
First pr is on staging. Once it's deployed we can move on to the second one

**Comment by stitesExpensify** [comment]
Same update, waiting for another deploy

**Comment by stitesExpensify** [comment]
Web PR is in review and should be merged shortly

**Comment by stitesExpensify** [comment]
Web PR was merged. Once it is on prod we can do the next /app pr

**Comment by stitesExpensify** [comment]
PHP is merged and on prod. Working on https://github.com/Expensify/App/pull/17996 this week (It has a lot of conflicts)

**Comment by stitesExpensify** [comment]
I'm not sure TBH. Let's put it on hold just in case since the changes are really big

**Comment by stitesExpensify** [comment]
Update: Fixed all conflicts and bugs on the /app PR. Hoping to get it merged asap to prevent more conflicts

**Comment by laurenreidexpensify** [comment]
@stitesExpensify how we doing on this one?

**Comment by laurenreidexpensify** [comment]
@stitesExpensify what's the latest here?

**Comment by stitesExpensify** [comment]
The main issue is no longer happening! 🎉  There were a couple of regressions that are getting taken care of, and then we can work on the clean-up which will complete the migration

**Comment by sakluger** [comment]
@stitesExpensify I just retested https://github.com/Expensify/App/issues/22061 and the behavior is different from before, but it's still not working as expected. Before, when emoji reacting to the "Completed task" or "Reopened task" message, it added the emoji reaction briefly but then it disappeared. Now, the emoji reaction never appears.

We thought that the fixes for this issue might fix that one too, but it hasn't. Do you think any of the cleanup tasks your working on here will fix that one? Or should we go ahead and make it external?

**Comment by stitesExpensify** [comment]
Replied in the other issue :)

**Comment by laurenreidexpensify** [comment]
@stitesExpensify is there an update on this one? Thanks

**Comment by stitesExpensify** [comment]
Still just cleaning up. I'm removing the Bug tag since the bug is fixed, we're just making the code better.

**Comment by stitesExpensify** [comment]
Cleanup has not moved forward at all, I've been pulled onto more important projects. Like I mentioned last week, the bug itself is fixed and not being tracked anymore. @laurenreidexpensify  If you want to follow along with the cleanup some more feel free, but otherwise you can un-assign :)

**Comment by stitesExpensify** [comment]
I've been focused on wave8. I should finally have some time to complete this very soon

**Comment by stitesExpensify** [comment]
no update

**Comment by stitesExpensify** [comment]
Closing for #focus on waves per the new guidelines

**Comment by iwiznia** [comment]
Reopening this because I see [a comment saying](https://github.com/Expensify/Web-Expensify/blob/9fdfb4a7b0f0ce5157a9b5b79762f2360ee3568e/api.php#L981) we will cleanup things and we did not. 
This would be a `finish what we started` task.

**Comment by stitesExpensify** [comment]
I have PRs up for both auth and web, but have not gotten them merged yet

**Comment by stitesExpensify** [comment]
no update

**Comment by neil-marcellini** [comment]
It's been like a year and a half and I just stumbled on this from server error issue. Will we please clean it up soon? About how long will it take to do so?

**Comment by stitesExpensify** [comment]
@neil-marcellini I wrote up some PRs but they have been shelved while I was focused on the travel deadline. I am definitely still aware that it needs to get done :)

**Comment by stitesExpensify** [comment]
There is a separate cleanup issue [here](https://github.com/Expensify/Expensify/issues/382289) that I will be tracking in
